lasagne/layers/cuda_convnet.py

```diff
@@ -198,34 +202,51 @@ def get_output_for(self, input, **kwargs):
 
 
 class MaxPool2DCCLayer(CCLayer):
-    def __init__(self, incoming, ds, ignore_border=False, strides=None,
+    def __init__(self, incoming, pool_size, ignore_border=False, stride=None,
                  dimshuffle=True, **kwargs):
         from pylearn2.sandbox.cuda_convnet.pool import MaxPool
 
+        if 'pad' in kwargs:
+            pad = kwargs.pop('pad')
+            if as_tuple(pad, 2) != (0, 0):
+                raise NotImplementedError("MaxPool2DCCLayer does not "
+                                          "support padding")
+
         super(MaxPool2DCCLayer, self).__init__(incoming, **kwargs)
-        if ds[0] != ds[1]:
-            raise RuntimeError("MaxPool2DCCLayer only supports square pooling "
-                               "regions, but ds=(%d, %d)" % ds)
 
-        if strides is not None and strides[0] != strides[1]:
-            raise RuntimeError("MaxPool2DCCLayer only supports using the same "
-                               "stride in both directions, but "
-                               "strides=(%d, %d)" % strides)
+        pool_size = as_tuple(pool_size, 2)
+
+        if pool_size[0] != pool_size[1]:
+            raise NotImplementedError("MaxPool2DCCLayer only supports square "
+                                      "pooling regions, but pool_size=(%d, %d)"
+                                      % pool_size)
+
+        self.pool_size = pool_size[0]
+
+        if stride is None:
+            self.stride = self.pool_size
+        else:
+            stride = as_tuple(stride, 2)
+            if stride[0] != stride[1]:
+                raise NotImplementedError("MaxPool2DCCLayer only supports "
+                                          "using the same stride in both, "
+                                          "directions but stride=(%d, %d)"
+                                          % stride)
+            self.stride = stride[0]
+
+        if self.stride > self.pool_size:
+            raise NotImplementedError("MaxPool2DCCLayer only supports "
+                                      "stride <= pool_size.")
 
         # ignore_border argument is for compatibility with MaxPool2DLayer.
         # it is not supported. Borders are never ignored.
         if ignore_border is not False:
-            raise RuntimeError("MaxPool2DCCLayer does not support "
-                               "ignore_border.")
+            raise NotImplementedError("MaxPool2DCCLayer does not support "
+                                      "ignore_border.")
```

Comment:
Makes sense. It's even something that one *could* add here if anybody ever needs it.